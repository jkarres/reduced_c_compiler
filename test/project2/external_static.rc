// just checking that these get initialized exactly once // you'll
// need to check the assembly yourself to see whether they're .global or
// not.

function : bool getTrue() {
  cout << "in getTrue()" << endl;
  return true;
}

static bool b = getTrue();

int i;

function : void main() {
  cout << "in main()" << endl;
  ++i;

  if (i == 1) { main(); }

}

