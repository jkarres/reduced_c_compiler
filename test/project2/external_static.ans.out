in getTrue()
in main()
in main()
